<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="cardboard">

    <xacro:property name="length" value="0.8" />
    <xacro:property name="width" value=".30" />
    <xacro:property name="thickness" value=".005" />
    <xacro:property name="joints" value="4" />

    <link name="input_flap">
    <visual>
        <geometry>
            <box size="${length/(joints+1)} ${width} ${thickness}"/>
        </geometry>
        <origin rpy="0 0 0" xyz="${0.5*length/(joints+1)} 0 0"/>
    </visual>
    </link>


        <link name="link_4">
            <visual>
                <geometry>
                    <box size="${length/(joints+1)} ${width} ${thickness}"/>
                </geometry>
                <origin rpy="0 0 0" xyz="${0.5*length/(joints+1)} 0 0"/>
            </visual>
        </link>

        <joint name="joint4" type="revolute">
            <axis xyz="0 -1 0"/>
            <limit effort="1000.0" lower="-2.96705973" upper="2.96705973" velocity="0.5"/>
            <origin rpy="0 0 0" xyz="${length/(joints+1)} 0.0 0"/>
            <parent link="input_flap"/>
            <child link="link_4"/>
        </joint>

        <link name="link_2">
            <visual>
                <geometry>
                    <box size="${length/(joints+1)} ${width} ${thickness}"/>
                </geometry>
                <origin rpy="0 0 0" xyz="${0.5*length/(joints+1)} 0 0"/>
            </visual>
        </link>

        <joint name="joint2" type="revolute">
            <axis xyz="0 -1 0"/>
            <limit effort="1000.0" lower="-2.96705973" upper="2.96705973" velocity="0.5"/>
            <origin rpy="0 0 0" xyz="${length/(joints+1)} 0.0 0"/>
            <parent link="link_4"/>
            <child link="link_2"/>
        </joint>

        <link name="link_3">
            <visual>
                <geometry>
                    <box size="${length/(joints+1)} ${width} ${thickness}"/>
                </geometry>
                <origin rpy="0 0 0" xyz="${0.5*length/(joints+1)} 0 0"/>
            </visual>
        </link>

        <joint name="joint3" type="revolute">
            <axis xyz="0 -1 0"/>
            <limit effort="1000.0" lower="-2.96705973" upper="2.96705973" velocity="0.5"/>
            <origin rpy="0 0 0" xyz="${length/(joints+1)} 0.0 0"/>
            <parent link="link_2"/>
            <child link="link_3"/>
        </joint>
    

        <link name="link_5">
            <visual>
                <geometry>
                    <box size="${length/(joints+1)} ${width} ${thickness}"/>
                </geometry>
                <origin rpy="0 0 0" xyz="${0.5*length/(joints+1)} 0 0"/>
            </visual>
        </link>

        <joint name="joint5" type="revolute">
            <axis xyz="0 -1 0"/>
            <limit effort="1000.0" lower="-2.96705973" upper="2.96705973" velocity="0.5"/>
            <origin rpy="0 0 0" xyz="${length/(joints+1)} 0.0 0"/>
            <parent link="link_3"/>
            <child link="link_5"/>
        </joint>

</robot>